akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -412,8 +414,10 @@ private[akka] object Running {
         // wait for snapshot response before stopping
         new StoringSnapshot(state.copy(receivedPoisonPill = true), sideEffects, snapshotReason)
       case signal =>
-        setup.onSignal(state.state, signal, catchAndLog = false)
-        Behaviors.same
+        if (setup.onSignal(state.state, signal, catchAndLog = false))
+          Behaviors.same
+        else
+          Behaviors.unhandled
```

Comment:
I looked at how to make it less boilerplatey but we wouldn't want to return any Behavior from `onSignal`, merely evaluate and handle the signal.